internal/cmd/codegen.go

```diff
@@ -0,0 +1,107 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"slices"
+	"strings"
+
+	"github.com/kynrai/gofs/internal/codegen"
+	"github.com/kynrai/gofs/internal/gofs"
+)
+
+const codegenUsage = `usage: gofs codegen [template] -struct=[struct name]
+
+Experimental: "codegen" generates code from go templates. 
+This should be used as a go:generate directive in the source code.
+
+Example:
+//go:generate gofs codegen db -struct=Foo
+type Foo struct {
+	ID   string  'json:"id"  gofs:"pk"'
+	Bar  string  'json:"bar" gofs:"searchable'
+}
+
+`
+
+func init() {
+	Gofs.AddCmd(Command{
+		Name:  "codegen",
+		Short: "generate go code from struct",
+		Long:  codegenUsage,
+		Cmd:   cmdCodegen,
+	})
+}
+
+func cmdCodegen() {
+	gofile := os.Getenv("GOFILE")
+	if gofile == "" {
+		fmt.Println("codegen: GOFILE not set")
+		return
+	}
+
+	gopackage := os.Getenv("GOPACKAGE")
+	if gopackage == "" {
+		fmt.Println("codegen: GOPACKAGE not set")
+		return
+	}
+
+	gostruct := ""
+	for _, arg := range os.Args {
+		if strings.HasPrefix(arg, "-struct=") {
```

Comment:
we actually parse args i think, so we can use the go built in args and flags to get the -struct